src/games/strategy/ui/SwingComponents.java

```diff
@@ -278,4 +285,50 @@ public static JMenu newJMenu(final String menuTitle, final KeyboardCode keyboard
     menu.setMnemonic(keyboardCode.getSwingKeyEventCode());
     return menu;
   }
+
+  /**
+   * Runs the specified task on a background thread while displaying a progress dialog.
+   *
+   * @param<T> The type of the task result.
+   *
+   * @param frame The {@code Frame} from which the progress dialog is displayed or {@code null} to use a shared, hidden
+   *        frame as
+   *        the owner of the progress dialog.
+   * @param message The message to display in the progress dialog; must not be {@code null}.
+   * @param task The task to be executed; must not be {@code null}.
+   *
+   * @return A promise that resolves to the result of the task; never {@code null}.
+   */
+  public static <T> CompletableFuture<T> runWithProgressBar(
+      final Frame frame,
```

Comment:
My bad.  I keep forgetting that the TripleA Eclipse formatter configuration does not join comment lines.  Will fix.